### PR TITLE
Feat: added the signup feature

### PR DESCRIPTION
**Pull request by rubymaharjan**
installed axios to fetch the API.
Worked on the signup feature.
communicated with API.

### CODE REVIEW

File: src/App.js [hunk 1]
```diff
@@ -1,26 +1,18 @@
...
-          target="_blank"
-          rel="noopener noreferrer"
-        >
-          Learn React
-        </a>
-      </header>
-    </div>
-  );
+    return (
+        <React.Fragment>
```
Comment (prashantacharya) [inline src/App.js]: It's better to use short hand syntax instead of <React.Fragment>

File: src/App.js [hunk 2]
```diff
@@ -1,26 +1,18 @@
...
+            <Switch>
+                <Route exact path='/' component={Landing} />
+                <Route path='/signin' component={SignUp} />
```
Comment (prashantacharya) [inline src/App.js]: **SignUp** component on **signin** route?

File: src/components/SignUp.jsx [hunk 3]
```diff
@@ -0,0 +1,83 @@
+import React, { Component } from "react";
+import mentors from "../images/mentor.jpg";
+import Login from "./Login";
+import SignUpForm from "./SignUpForm";
+export default class SignUp extends Component {
+  tabHandler = (evt, cityName) => {
+    var i, tabcontent, tablinks;
```
Comment (prashantacharya) [inline src/components/SignUp.jsx]: can we use **let** instead of **var**?

File: src/components/SignUpForm.jsx [hunk 4]
```diff
@@ -0,0 +1,77 @@
...
+
+  handleSubmit = e => {
+    e.preventDefault();
+    // const data = {
+    //   first_name: this.state.first_name,
+    //   last_name: this.state.last_name,
+    //   email: this.state.email,
+    //   password: this.state.password
+    // };
+    axios
```
Comment (prashantacharya) [inline src/components/SignUpForm.jsx]: we could try using **async await** here.

File: src/App.js (hunk 2 above)
Comment (shakyasaijal) [reply src/App.js]: @rubymaharjan

File: public/index.html [hunk 5]
```diff
@@ -5,34 +5,14 @@
...
-      Notice the use of %PUBLIC_URL% in the tags above.
-      It will be replaced with the URL of the `public` folder during the build.
-      Only files inside the `public` folder can be referenced from the HTML.
-
-      Unlike "/favicon.ico" or "favicon.ico", "%PUBLIC_URL%/favicon.ico" will
-      work correctly both with client-side routing and a non-root public URL.
-      Learn how to configure a non-root public URL by running `npm run build`.
-    -->
+    <!-- <script src="./template.js"></script> -->
     <title>React App</title>
```
Comment (Xkid0525) [inline public/index.html]: Need to change the title of the app

File: src/components/Login.jsx [hunk 6]
```diff
@@ -0,0 +1,13 @@
+import React, { Component } from 'react'
+
+export default class Login extends Component {
+    render() {
+        return (
+            <form onSubmit={this.handleSubmit} method="post">
```
Comment (shakyasaijal) [inline src/components/Login.jsx]: Cover form with div

File: src/components/Landing.jsx [hunk 7]
```diff
@@ -0,0 +1,7 @@
+import React, { Component } from "react";
+
+export default class Landing extends Component {
+  render() {
+    return <div>Landing</div>;
```
Comment (Xkid0525) [inline src/components/Landing.jsx]: what is the use of this component when you only have a single tag with just text??

File: src/components/SignUpForm.jsx [hunk 8]
```diff
@@ -0,0 +1,77 @@
+import React, { Component } from "react";
+import axios from "axios";
+export default class SignUpForm extends Component {
+  state = {
+    first_name: "",
+    last_name: "",
+    email: "",
+    password: ""
+  };
+  //   componentDidMount() {
```
Comment (shakyasaijal) [inline src/components/SignUpForm.jsx]: Remove unnecessary comments @rubymaharjan

File: src/components/SignUp.jsx (hunk 3 above)
Comment (Xkid0525) [reply src/components/SignUp.jsx]: you should either use just es5 or just es6, try not to mix things up

you can use let

File: src/components/SignUp.jsx [hunk 9]
```diff
@@ -0,0 +1,83 @@
...
+    tabcontent = document.getElementsByClassName("tabcontent");
+    for (i = 0; i < tabcontent.length; i++) {
+      tabcontent[i].style.display = "none";
+    }
+    tablinks = document.getElementsByClassName("tablinks");
+    for (i = 0; i < tablinks.length; i++) {
+      tablinks[i].className = tablinks[i].className.replace(" active", "");
+    }
+    document.getElementById(cityName).style.display = "block";
```
Comment (Xkid0525) [inline src/components/SignUp.jsx]: i guess the correct syntax is document.getElementById("cityName")
or is there a specific reason you used it without quotes?

File: src/components/SignUpForm.jsx [hunk 10]
```diff
@@ -0,0 +1,77 @@
...
+  handleSubmit = e => {
+    e.preventDefault();
+    // const data = {
+    //   first_name: this.state.first_name,
+    //   last_name: this.state.last_name,
+    //   email: this.state.email,
+    //   password: this.state.password
+    // };
+    axios
+      .post("http://192.168.1.125:3000/mentors/signup", this.state)
```
Comment (shakyasaijal) [inline src/components/SignUpForm.jsx]: Use proxy instead of hard coded url

Comment (shakyasaijal) [reply src/components/SignUpForm.jsx]: @rubymaharjan

File: src/components/Login.jsx (hunk 6 above)
Comment (rubymaharjan) [reply src/components/Login.jsx]: The form itself encloses all the content. Is it still necessary?

File: src/App.js (hunk 1 above)
Comment (rubymaharjan) [reply src/App.js]: ok

File: src/components/SignUp.jsx (hunk 3 above)
Comment (rubymaharjan) [reply src/components/SignUp.jsx]: yes. ok

File: src/components/Authentications/Login.jsx [hunk 11]
```diff
@@ -0,0 +1,36 @@
+import React, { Component } from 'react'
+
+export default class Login extends Component {
+    render() {
+        return (
+            <>
```
Comment (Xkid0525) [inline src/components/Authentications/Login.jsx]: What is this?

**Comment by khadkarajesh** [comment]
@shakyasaijal @Xkid0525  @Prashant-Acharya  please have a  look once again.

File: src/App.js [hunk 12]
```diff
@@ -1,26 +1,13 @@
...
-          target="_blank"
-          rel="noopener noreferrer"
-        >
-          Learn React
-        </a>
-      </header>
-    </div>
-  );
+    return (
+        <>
```
Comment (shakyasaijal) [inline src/App.js:10]: Is it ok to use just <>?

File: src/components/Authentications/Login.jsx (hunk 11 above)
Comment (shakyasaijal) [reply src/components/Authentications/Login.jsx]: @rubymaharjan  @Prashant-Acharya

File: src/static/template.js [hunk 13]
```diff
@@ -0,0 +1,15 @@
+const tabHandler = (evt, cityName) => {
+  let i, tabcontent, tablinks;
```
Comment (shakyasaijal) [inline src/static/template.js:2]: Use camel case.

**Pull request merged**